jujugui/static/gui/src/app/components/deployment/summary/test-summary.js

```diff
@@ -71,11 +71,26 @@ describe('DeploymentSummary', function() {
         change={changeDescriptions[1]} />];
     var renderer = jsTestUtils.shallowRender(
       <juju.components.DeploymentSummary
-        jem={{}}
+        jem={{
+          listTemplates: cb => {
+            cb(null, [{
+              path: 'spinach/my-creds',
+              location: { cloud: 'aws', region: 'us-west-1' }
+            }]);
+          },
+          listRegions: (cloud, cb) => {
+            assert.equal(cloud, 'aws');
+            cb(null, ['us-west-1', 'us-east-1']);
+          }
+        }}
```

Comment:
We've got JS nested in JSX nested in JS here. If it's much more than a simple expression, I'd rather see it setup outside the JSX, assigned to a variable, and that keeps the JSX more presentation-focused and HTML-like.